[gamemodes]/[play]/play/play_vehicles.lua

```diff
@@ -0,0 +1,98 @@
+local vehicleTimers = {}
+local playerVehicles = {}
+local vehiclesToSpawn = {}
+
+function createVehicles()
+	for vehicleID = 1, #vehicleSpawns do
+		createPlayVehicle(vehicleSpawns[vehicleID])
+	end
+end
+
+function createPlayVehicle(vehicleData)
+	local modelID, posX, posY, posZ, rotX = vehicleData[1], vehicleData[2], vehicleData[3], vehicleData[4], vehicleData[5]
+	local vehicleElement = createVehicle(modelID, posX, posY, posZ, 0, 0, rotX)
+
+	setVehicleDamageProof(vehicleElement, true)
+	setElementFrozen(vehicleElement, true)
+	vehiclesToSpawn[vehicleElement] = vehicleData
+end
+
+function destroyVehicle(vehicleElement)
+	if isElement(vehicleElement) then
+		destroyElement(vehicleElement)
+	end
+
+	destroyVehicleTimer(vehicleElement)
+end
+
+function destroyPlayerVehicles(playerElement)
+	local savedVehicles = playerVehicles[playerElement]
+
+	if savedVehicles then
+
+		for vehicleID = 1, #savedVehicles do
+			local vehicleElement = savedVehicles[vehicleID]
+
+			if isElement(vehicleElement) then
+				destroyElement(vehicleElement)
+			end
+
+			destroyVehicleTimer(vehicleElement)
+		end
+
+		playerVehicles[playerElement] = nil
+	end
+end
+
+function destroyVehicleTimer(vehicleElement)
+	local vehicleTimer = vehicleTimers[vehicleElement]
+
+	if vehicleTimer then
+
+		if isTimer(vehicleTimer) then
+			killTimer(vehicleTimer)
+		end
+
+		vehicleTimers[vehicleElement] = nil
+	end
+end
+
+function onVehicleEnter(playerElement)
+	local vehicleRespawnTime = get("vehicleRespawnTime")
+	local vehicleData = vehiclesToSpawn[source]
+
+	vehicleRespawnTime = tonumber(vehicleRespawnTime) or 60000
+	vehicleRespawnTime = vehicleRespawnTime > 50 and vehicleRespawnTime or 50
```

Comment:
Timers support shorter durations than 50ms nowadays.
```suggestion
	vehicleRespawnTime = vehicleRespawnTime > 0 and vehicleRespawnTime or 0
```